AIDA: [DOC-3957] updated overview section

rkapoor10 opened the pull request:
## Summary   
- updated `docs/platform/Harness-AIDA/aida-overview.md`
deploy preview : https://65a7c3dd319d870525b03b02--harness-developer.netlify.app/docs/platform/Harness-AIDA/aida-overview
<img width="978" alt="Screenshot 2024-01-17 at 5 39 37 PM" src="https://github.com/harness/developer-hub/assets/61417822/60ddcd5f-02a8-4cde-8769-6e8d68eb8093">

## What Type of PR is This?

- [ ] Issue
- [x] Feature Update
- [ ] Maintenance/Chore


## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [X] Tested Locally
- [X] *Optional* Screen Shoot.

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a7c3dd319d870525b03b02--harness-developer.netlify.app

brian-f-harness commented on docs/platform/Harness-AIDA/aida-overview.md:
```diff
@@ -22,20 +21,35 @@ Here are some key benefits of Harness AIDA:
+| Automate feature flags integration                                                                                               | Incorporate feature flags into your development process through tailored code generation for seamless integration. | FF       | Beta                |
+| Dashboard Intelligence                                                                                                           | AIDA empowers you to craft customized dashboards with widget-level control through interactive prompts.            | Platform | Beta                |
+| Policy As Code Assistant                                                                                                         | Generate and integrate Open Policy Agent (OPA) Rego policies to meet your compliance standards.                    | CD       | Development         |
+| Semantic Search                                                                                                                  | Search code repositories to provide accurate and context-aware results in Harness CODE.                            | CODE     | Development         |
```
```suggestion
| Semantic Search                                                                                                                  | Search code repositories to provide accurate and context-aware results in Harness Code.                            | CODE     | Development         |
```

brian-f-harness commented on docs/platform/Harness-AIDA/aida-overview.md:
```diff
@@ -22,20 +21,34 @@ Here are some key benefits of Harness AIDA:
+| Feature                                                                                                                          | Description                                                                                                        | Module   | Status              |
+| -------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ | -------- | ------------------- |
+| [Troubleshoot CI builds](/docs/continuous-integration/troubleshoot-ci/aida)                                                      | Resolve your build failures with AIDA's root cause analysis (RCA).                                                 | CI       | Generally available |
+| Troubleshoot CD deployments                                                                                                      | Resolve your deployment failures with AIDA's root cause analysis (RCA).                                                 | CD       | Generally available |
```
```suggestion
| Troubleshoot CD deployments                                                                                                      | Resolve your deployment failures with AIDA's RCA.                                                 | CD       | Generally available |
```

brian-f-harness commented on docs/platform/Harness-AIDA/aida-overview.md:
```diff
@@ -22,20 +21,34 @@ Here are some key benefits of Harness AIDA:
+| Feature                                                                                                                          | Description                                                                                                        | Module   | Status              |
+| -------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ | -------- | ------------------- |
+| [Troubleshoot CI builds](/docs/continuous-integration/troubleshoot-ci/aida)                                                      | Resolve your build failures with AIDA's root cause analysis (RCA).                                                 | CI       | Generally available |
+| Troubleshoot CD deployments                                                                                                      | Resolve your deployment failures with AIDA's root cause analysis (RCA).                                                 | CD       | Generally available |
```
RCA is defined in the first instance, can use the acronym here.

rkapoor10 replied:
Sure, thanks for the review!

aimurphy commented:
I am making some changes to the Early access page to reflect the content in the table. I will push my change here.

rkapoor10 commented:
> I am making some changes to the Early access page to reflect the content in the table. I will push my change here.

Sure April, will hold this one then.

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a7ef0da8abd30c8be33c64--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a7ef21af11950844f11d64--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a7f077d5b5f8101b3fc138--harness-developer.netlify.app

aimurphy commented on docs/platform/Harness-AIDA/aida-overview.md:
```diff
@@ -22,20 +21,34 @@ Here are some key benefits of Harness AIDA:
 ## AIDA in Harness modules
 
-Currently, AIDA is available for CCM, CI, and STO.
+| Feature                                                                                                                          | Description                                                                                                        | Module   | Status              |
```
I am going to reorganize the table to be module-first. It is more likely that users will scan for the modules they use first, rather than by feature name.

rkapoor10 replied:
Okay, maybe 'Feature - Module - Description - Status' should also work. I added Module at the start, but it was looking a bit off. It's your call...

aimurphy replied:
Could you explain what you mean by it looked a bit off?

Trying to think from the user's perspective, I want to know the latest and greatest for my modules, so I want to skim for my modules' names first, and then read about the features for those modules.

rkapoor10 replied:
@aimurphy should we keep the table sorted based on status ?

aimurphy replied:
I think alphabetical-by-module is best because it keeps the module features together. Then a user can, for example, see all the CD features (both GA and beta) at once.

If you want them separate, I would create multiple tables or lists for each status.

rkapoor10 replied:
Sure, let's keep it as is for now. We can separate it out once the list grows with more modules adopting AIDA 💪

aimurphy commented:
@rkapoor10 : I updated the Early access page, and I had to make some corrections for CD and CCM early access features as well. Aside from moving the AIDA items, this was unrelated to your change.

I also removed redundancies on the AIDA overview page and reorganized the table to be more easily browsed by module name.

brian-f-harness commented on docs/platform/Harness-AIDA/aida-overview.md:
```diff
@@ -4,37 +4,26 @@ description: Learn about how AIDA improves your experience on the Harness platfo
+| CD | Policy As Code Assistant | Generate and integrate Open Policy Agent (OPA) Rego policies to meet your compliance standards. | Development |
+| CE | ChaosGuard | Generate conditions for your chaos experiments with ChaosGuard. | Generally available |
+| CODE | Semantic Search | Search code repositories to provide accurate and context-aware results in Harness Code. | Development |
+| CI | [Troubleshoot CI builds](/docs/continuous-integration/troubleshoot-ci/aida) | Resolve your build failures with AIDA's root cause analysis (RCA) | Generally available |
```
```suggestion
| CI | [Troubleshoot CI builds](/docs/continuous-integration/troubleshoot-ci/aida) | Resolve your build failures with AIDA's RCA. | Generally available |
```

rkapoor10 commented:
> @rkapoor10 : I updated the Early access page, and I had to make some corrections for CD and CCM early access features as well. Aside from moving the AIDA items, this was unrelated to your change.
> 
> I also removed redundancies on the AIDA overview page and reorganized the table to be more easily browsed by module name.

ack, need a final approval to merge this. Thanks

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a7f705b9659a0563556d86--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a7f9d3f7f5f017b286fd55--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a96601c9126000963c4da9--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://65a96725edeba30327786344--harness-developer.netlify.app
